### PR TITLE
refactors gloves/mask/glasses/handcuff dispensers

### PR DESCRIPTION
**Pull request by Adharainspace**
[CLEANLINESS]
## About the PR 

- adds a parent object for item dispensers so theres a lot less duplicate code

- moves item dispensers out of obj.dm

- changes map references to match the new types

## Why's this needed? 
there was lots of weird/duplicate code, it was hell to use them for mapping (they were all over the tree, not grouped under a parent) and they were organisationally bad. this fixes that!

## Changelog

```
```

### CODE REVIEW

File: code/obj/item_dispensers.dm [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+/obj/item_dispenser
+	name = "item dispenser"
+	desc = "A storage container that easily dispenses items."
+	icon = 'icons/obj/stationobjs.dmi'
+	pixel_y = 28
+	var/filled_icon_state = "" //i tried to do this in a smart way but it was a PITA so here have this stinky code instead
+	var/empty_icon_state = "" //autoset by the s y s t e m, dont set this yourself
+	var/amount = 3 //how many items does it have?
+	var/deposit_type = null //this is a type that this item will accept to "reload" itself
+	var/withdraw_type = null //this is a type that this item will dispense
```
Comment (ZeWaka) [inline code/obj/item_dispensers.dm:14]: wow nice!

File: code/obj/item_dispensers.dm [hunk 2]
```diff
@@ -0,0 +1,87 @@
+//////////
+//PARENT//
+//////////
+
+/obj/item_dispenser
+	name = "item dispenser"
+	desc = "A storage container that easily dispenses items."
+	icon = 'icons/obj/stationobjs.dmi'
```
Comment (ZeWaka) [inline code/obj/item_dispensers.dm:8]: should have a default icon_state

**Comment by Adharainspace** [comment]
i added a default icon state

File: code/obj/item_dispensers.dm [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	desc = "A storage container that easily dispenses medical masks."
+	icon_state = "dispenser_mask"
+	filled_icon_state = "dispenser_mask"
+	deposit_type = /obj/item/clothing/mask/medical
+	withdraw_type = /obj/item/clothing/mask/medical
+
+/obj/item_dispenser/perscription_glasses
+	name = "perscription glasses dispenser"
+	desc = "A storage container that easily dispenses perscription glasses."
+	icon_states = "dispenser_glasses"
```
Comment (flrsh) [inline code/obj/item_dispensers.dm:89]:
```suggestion
	icon_state = "dispenser_glasses"
```

Comment (Adharainspace) [reply code/obj/item_dispensers.dm:89]: this is good, i think,

Comment (Adharainspace) [reply code/obj/item_dispensers.dm:89]: cant automerge this on mobile ew, 1 minute

**Pull request merged**